rpc/src/rpc.rs

```diff
@@ -940,11 +1050,11 @@ impl JsonRpcRequestProcessor {
             })
         } else {
             let (addresses, address_filter) = if let Some(filter) = config.clone().filter {
-                let non_circulating_supply =
-                    calculate_non_circulating_supply(&bank).map_err(|e| {
-                        RpcCustomError::ScanError {
-                            message: e.to_string(),
-                        }
+                let non_circulating_supply = self
+                    .calculate_non_circulating_supply(&bank)
+                    .await
+                    .map_err(|e| RpcCustomError::ScanError {
+                        message: e.to_string(),
```

Comment:
This looks good! Below where we call bank.get_largest_accounts - that one
probably needs to go in bg too? It does an account scan which I think is slow by
definition

Reply:
Can do. Maybe more concerning is that the `largest_accounts_cache` doesn't appear to me to be a read-through cache. I think that given an empty cache, multiple callers may perform that fetch many times, rather than all locking on a single read.